### PR TITLE
New resource leak unit test for 

### PR DESCRIPTION
**Pull request by xi-liu-ds**
In this PR, we add a new unit test to reflect the [fix](https://github.com/facebook/infer/pull/1601) we had in backend, in which now an interprocedural cleanup function can close the dangling global resource. Such as the following example shows:

image

### CODE REVIEW

File: Cilsil.Test/Assets/TestClass.cs [hunk 1]
```diff
@@ -37,14 +37,9 @@ public struct TestStruct
 
         public TestClass() { }
 
-        public TestClass(TestClass testClass)
+        public TestClass(string filename)
```
Comment (matjin) [inline Cilsil.Test/Assets/TestClass.cs:40]: I think it's better to actually create an optional string parameter (default null); if a non-null string is provided, the InstanceStreamReaderField is initialized with the string.

Comment (xi-liu-ds) [reply Cilsil.Test/Assets/TestClass.cs:40]: > I think it's better to actually create an optional string parameter (default null); if a non-null string is provided, the InstanceStreamReaderField is initialized with the string.

I found that the function invoked `TestClass(TestClass testClass)` (`InvokeConstructorWithNullParam`) was not used anywhere. So I removed `InvokeConstructorWithNullParam` function and reused the constructor as `TestClass(string filename)`.

Comment (matjin) [reply Cilsil.Test/Assets/TestClass.cs:40]: I'm pretty sure this method was added (with the method InvokeConstructorWithNullParam()) to provide coverage over a case in which we were providing arguments in a flipped order, and where we were thus seeing cases where "this" was null. The methods of TestClass are often not for actual explicit invocation (as you see with InitializeInstanceObjectField, for example) but to provide coverage over certain constructs. I don't think we should remove this case -- i believe it was included to validate against a regression.

Comment (xi-liu-ds) [reply Cilsil.Test/Assets/TestClass.cs]: Done. Not exactly like you suggested though. If we do it like you suggested, Infer will think it is possible that `string filename` is not null and there will be a resource leak inside `InvokeConstructorWithNullParam()`.

**Review by matjin — DISMISSED** [review]
thank you -- looks good!

**Review by matjin — APPROVED** [review]

**Pull request merged**